source/Missions/Services/Network/NetworkAgentRegistry.cs

```diff
@@ -106,28 +167,32 @@ public bool RemoveControlledAgent(Guid agentId)
             return false;
         }
 
-        public bool RemoveNetworkControlledAgent(Guid agentId)
+        public bool RemovePlayerAgent(Guid agentId)
         {
-            return m_OtherAgents.Values.Any(group =>
+            if (m_PlayerAgents.TryGetValue(agentId, out Agent agent))
             {
-                Agent agent = group.RemoveAgent(agentId);
-                if (agent != null)
-                {
-                    return m_AgentToId.Remove(agent);
-                }
-                else
-                {
-                    return false;
-                }
-            });
+                return m_AgentToId.Remove(agent);
+            }
+
+            return false;
```

Comment:
Change this method to only be `return m_PlayerAgents.Remove(agentId)` all the other stuff is not needed